lib/ansible/modules/files/xml.py

```diff
@@ -0,0 +1,757 @@
+            xmlstring=dict(type='str'),
+            xpath=dict(type='str', default='/'),
+            namespaces=dict(type='dict', default={}),
+            state=dict(type='str', default='present', choices=['absent', 'present'], aliases=['ensure']),
+            value=dict(),
+            attribute=dict(),
+            add_children=dict(type='list'),
+            set_children=dict(type='list'),
+            count=dict(type='bool', default=False),
+            print_match=dict(type='bool', default=False),
+            pretty_print=dict(type='bool', default=False),
+            content=dict(type='str', choices=['attribute', 'text']),
+            input_type=dict(type='str', default='yaml', choices=['xml', 'yaml'])
+        ),
+        supports_check_mode=True,
+        mutually_exclusive=[
+            ['value', 'set_children'],
+            ['value', 'add_children'],
+            ['set_children', 'add_children'],
+            ['path', 'xmlstring'],
+            ['content', 'set_children'],
+            ['content', 'add_children'],
+            ['content', 'value'],
+        ]
+    )
+
+    xml_file = module.params['path']
+    xml_string = module.params['xmlstring']
+    xpath = module.params['xpath']
+    namespaces = module.params['namespaces']
+    state = module.params['state']
+    value = json_dict_bytes_to_unicode(module.params['value'])
+    attribute = module.params['attribute']
+    set_children = json_dict_bytes_to_unicode(module.params['set_children'])
+    add_children = json_dict_bytes_to_unicode(module.params['add_children'])
+    pretty_print = module.params['pretty_print']
+    content = module.params['content']
+    input_type = module.params['input_type']
+    print_match = module.params['print_match']
+    count = module.params['count']
+
+    # Check if we have lxml 2.3.0 or newer installed
+    if not HAS_LXML:
+        module.fail_json(msg='The xml ansible module requires the lxml python library installed on the managed machine')
+    elif LooseVersion('.'.join(map(str, etree.LXML_VERSION))) < LooseVersion('2.3.0'):
+        module.fail_json(msg='The xml ansible module requires lxml 2.3.0 or newer installed on the managed machine')
+    elif LooseVersion('.'.join(map(str, etree.LXML_VERSION))) < LooseVersion('3.0.0'):
+        module.warn('Using lxml version lower than 3.0.0 does not guarantee predictable element attribute order.')
+
+    # Check if the file exists
+    if xml_string:
+        infile = BytesIO(xml_string.encode('utf-8'))
+    elif os.path.isfile(xml_file):
+        infile = open(xml_file, 'r')
+    else:
+        module.fail_json(msg="The target XML source does not exist: %s" % xml_file)
+
+    # Try to parse in the target XML file
+    try:
+        parser = etree.XMLParser(remove_blank_text=pretty_print)
+        x = etree.parse(infile, parser)
+    except etree.XMLSyntaxError:
+        e = get_exception()
+        module.fail_json(msg="Error while parsing path: %s" % e)
```

Comment:
Not necessary but exception handling (since this is ansible-2.4+) cna be written as:
```
    except etree.XMLSyntaxError as e:
        module.fail_json(msg="Error while parsing path: %s" % to_native(e))
```